### PR TITLE
go: Define a go version for building packages

### LINKED ISSUE #206: go-based packages should all be built with same golang version
We're currently hardcoding the version of golang used to build go-based packages:

```
$ grep -iIrn go_version | grep -v tar
proxy/cc-proxy.spec-template:7:%global GO_VERSION 1.8.3
.ci/setup.sh:26:go_version=1.7
runtime/cc-runtime.spec-template:7:%global GO_VERSION 1.8.3
```

As shown, we're also using two different versions.

The packaging script should use a single golang version for building all golang packages *and* for the CI. Ideally, the packaging script would use the version defined as `go_version=` in:

- https://github.com/clearcontainers/runtime/blob/master/versions.txt

But if that's not practical, we need to add `go_version=` to https://github.com/clearcontainers/packaging/blob/master/versions.txt.

### PR DESCRIPTION
**Pull request by erick0z**
Right now, the go version is hardcoded in the scripts. This commit adds
a go_version in the versions.txt file with the latest go release
version. This variable is now used across projects in order to build
them with the same go version.

Fixes #206

### CODE REVIEW

File: runtime/update_runtime.sh [hunk 1]
```diff
@@ -39,47 +39,56 @@ PROJECT_REPO=${PROJECT_REPO:-home:clearcontainers:clear-containers-3-staging/cc-
...
     sed -e "s/@VERSION@/$VERSION/g" \
-	    -e "s/@RELEASE@/$RELEASE/g" \
-	    -e "s/@HASH@/$short_hashtag/g" \
-	    -e "s/@cc_proxy_version@/$proxy_obs_fedora_version/" \
-	    -e "s/@cc_shim_version@/$shim_obs_fedora_version/" \
-	    -e "s/@cc_image_version@/$image_obs_fedora_version/" \
-	    -e "s/@linux_container_version@/$linux_container_obs_fedora_version/" \
-	    -e "s/@qemu_lite_obs_fedora_version@/$qemu_lite_obs_fedora_version/g" \
-	    cc-runtime.spec-template > cc-runtime.spec
+	-e "s/@RELEASE@/$RELEASE/g" \
```
Comment (devimc) [inline runtime/update_runtime.sh]: I think this shouldn't be part of the commit

File: runtime/update_runtime.sh [hunk 2]
```diff
@@ -39,47 +39,56 @@ PROJECT_REPO=${PROJECT_REPO:-home:clearcontainers:clear-containers-3-staging/cc-
...
+	-e "s/@HASH@/$short_hashtag/g" \
+	-e "s/@cc_proxy_version@/$proxy_obs_fedora_version/" \
+	-e "s/@cc_shim_version@/$shim_obs_fedora_version/" \
+	-e "s/@cc_image_version@/$image_obs_fedora_version/" \
+	-e "s/@linux_container_version@/$linux_container_obs_fedora_version/" \
+	-e "s/@qemu_lite_obs_fedora_version@/$qemu_lite_obs_fedora_version/g" \
+        -e "s/@GO_VERSION@/$go_version/" \
```
Comment (devimc) [inline runtime/update_runtime.sh]: bad indentation ?

File: runtime/update_runtime.sh [hunk 3]
```diff
@@ -39,47 +39,56 @@ PROJECT_REPO=${PROJECT_REPO:-home:clearcontainers:clear-containers-3-staging/cc-
...
+	cc-runtime.spec-template > cc-runtime.spec
 
     sed -e "s/@VERSION@/$VERSION/" \
-	    -e "s/@HASH@/$short_hashtag/" debian.rules-template > debian.rules
+	-e "s/@HASH@/$short_hashtag/" \
+        -e "s/@GO_VERSION@/$go_version/" \
```
Comment (devimc) [inline runtime/update_runtime.sh]: and here

File: versions.txt [hunk 4]
```diff
@@ -9,6 +9,10 @@ clear_vm_kernel_version=4.9.60
 clear_vm_image_version=19490
 qemu_lite_version=2.7.1
 
+# Golang
+go_version=1.9.2
+go_checksum=de874549d9a8d8d8062be05808509c09a88a248e77ec14eb77453530829ac02b
```
Comment (jodh-intel) [inline versions.txt]: Could you just add a comment explaining this. Something like:

```
# sha256 checksum for the go_version binary distribution ("go${go_version}.linux-amd64.tar.gz")
go_checksum=de874549d9a8d8d8062be05808509c09a88a248e77ec14eb77453530829ac02b
```

**Review by jodh-intel — COMMENTED** [review]
Great! Just one comment ;)

**Comment by erick0z** [comment]
@jodh-intel :)

**Comment by jodh-intel** [comment]
Thanks @erick0z!

lgtm

**Comment by jodh-intel** [comment]
Hi @jcvenegas - could you take a look at this?

**Review by jcvenegas — APPROVED** [review]
